core/app_client.go

```diff
@@ -60,16 +61,113 @@ func (a appClient) GetSurveyResponse(id string, orgID string, appID string, user
 	return a.app.storage.GetSurveyResponse(id, orgID, appID, userID)
 }
 
-// GetSurveyResponses returns the survey responses matching the provided filters
-func (a appClient) GetSurveyResponses(orgID string, appID string, userID string, surveyIDs []string, surveyTypes []string, startDate *time.Time, endDate *time.Time, limit *int, offset *int) ([]model.SurveyResponse, error) {
+// GetUserSurveyResponses returns the survey responses matching the provided filters for a specific user
+func (a appClient) GetUserSurveyResponses(orgID string, appID string, userID string, surveyIDs []string, surveyTypes []string, startDate *time.Time, endDate *time.Time, limit *int, offset *int) ([]model.SurveyResponse, error) {
 	return a.app.storage.GetSurveyResponses(&orgID, &appID, &userID, surveyIDs, surveyTypes, startDate, endDate, limit, offset)
 }
 
+// GetAllSurveyResponses returns the survey responses matching the provided filters
+func (a appClient) GetAllSurveyResponses(orgID string, appID string, userID string, surveyID string, startDate *time.Time, endDate *time.Time, limit *int, offset *int, externalIDs map[string]string) ([]model.SurveyResponse, error) {
+	var allResponses []model.SurveyResponse
+	var err error
+
+	survey, err := a.app.shared.getSurvey(surveyID, orgID, appID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if user is admin of calendar event if survey associated with one
+	if len(survey.CalendarEventID) > 0 {
```

Comment:
I think if the `survey.CalendarEventID` is not empty we would want to immediately return nil/an error. Currently I believe this function would allow anyone to retrieve the survey responses of any non-event survey which we definitely need to prevent.